File: Python/7py 1/f.py
"""
|--------------------------------------------------------------------------
| Gauß needs help! (Sums of a lot of numbers).
|--------------------------------------------------------------------------
|
"""

def f(n):
   return None if type(n) != int or n % 1 != 0 or n <= 0 else ((n + 1) * n) / 2


def f(n):
   return int(n * (n + 1) / 2)


def f(n):
   return sum(range(1, n + 1)) if type(n) is int and n > 0 else None


def f(n):
   return ((n+1)/2)*n if isinstance(n,int) and n > 0 else None


print(f(1), 1)
print(f(100), 5050)

File: PHP/7php 1/f.php
<?php
/*
|--------------------------------------------------------------------------
| Gauß needs help! (Sums of a lot of numbers).
|--------------------------------------------------------------------------
|
*/

function f($n = false) {
   return gettype($n) != 'integer' || $n % 1 != 0 || $n <= 0 ? false : (($n + 1) * $n) / 2;
}


function f($n = false) {
   $res = 0;
   if (gettype($n) == 'integer' && $n % 1 == 0 && $n > 0) {
      for ($i = 1; $i <= $n; $i++) {
         $res += $i;
      }
      return $res;
   } else {
      return false;
   }
}


function f($n = 0) {
   return is_int($n) && $n > 0 ? $n * ($n + 1) / 2 : false;
}


function f($n = false) {
   return (is_int($n) && $n >= 1)?array_sum(range(1, $n)):false;
}


print_r(f(100) . ' ' . 5050 . '
');
print_r(f('we'));
print_r(f());